src/builder.rs

```diff
@@ -88,6 +91,14 @@ impl<S: LogStorage, P: Processor<S::Transaction>> IndexerBuilder<S, P> {
         self
     }
 
+    pub fn finality_level(mut self, level: BlockNumberOrTag) -> Self {
+        if let BlockNumberOrTag::Number(_) = level {
```

Comment:
I am thinking that we can maybe create our own duplication of enum for this input.
I think something like
```rust
pub enum FinalityLevel {
        Pending,
        Finalized,
        .....
}
```

I think this can have an Into<BlockNumberOrTag> and make it more clear when we are passing parameter, as BlockNumberOrTag is slightly designed to describe different thing + this keeps us less attached to alloy in case they change anything and will allow to add serialization on this type later to be able to conviniently use in configs